### PR TITLE
make s3 calls in parallel

### PR DESCRIPTION
**Pull request by tanonev**
ptal @cliu587 @zhaojunz

### CODE REVIEW

File: aegisthus-distcp/src/main/java/Distcp.java [hunk 1]
```diff
@@ -215,32 +219,51 @@ protected void setupTempDirs() {
 	}
 
 	protected void setReducers(Job job, int fileCount) {
-		int reducers = job.getConfiguration().getInt("mapred.reduce.tasks", 1);
+		int reducers = job.getConfiguration().getInt("mapred.job.reduces", 1);
```
Comment (cliu587) [inline aegisthus-distcp/src/main/java/Distcp.java]: why do we need to change this?

**Comment by cliu587** [comment]
LGTM, other than my question on why the config name change.

File: aegisthus-distcp/src/main/java/Distcp.java (hunk 1 above)
Comment (tanonev) [inline aegisthus-distcp/src/main/java/Distcp.java]: The former is deprecated, and for some reason it was set to 23 when I ran it on the long-running cluster, while the non-deprecated one is set to 1, which is consistent with the defaults in hadoop's documentation.  If the deprecated and non-deprecated configuration fields can simultaneously store distinct values, we need to be better about avoiding deprecated fields.

File: aegisthus-distcp/src/main/java/Distcp.java [hunk 2]
```diff
@@ -215,32 +219,51 @@ protected void setupTempDirs() {
...
+						}
+						return curInputs;
+					}
+				}));
+			}
+			for (Future<List<String>> inputFuture : inputFutures) {
+				try {
+					inputs.addAll(inputFuture.get());
+				} catch(Exception e) {
+					e.printStackTrace();
```
Comment (danchia) [inline aegisthus-distcp/src/main/java/Distcp.java]: should we be swallowing the exception? Won't an exception mean we'd be missing some files potentially, and swallowing mean we're silently dropping some data?

**Pull request merged**